### PR TITLE
exoscale-cli: init at 1.12.0

### PR DESCRIPTION
**Pull request by dramaturg**
###### Motivation for this change

Add a derivation the Exoscales cli tool

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:10]:
```suggestion
  src = fetchFromGitHub {
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:1]:
```suggestion
{ stdenv, buildGo114Package, substituteAll, fetchFromGitHub }:
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:3]:
```suggestion
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:4]:
```suggestion
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:5]:
```suggestion
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:6]:
```suggestion
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:7]: Why not `buildGoPackage`?

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:11]:
```suggestion
    owner = "exoscale";
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:8]:
```suggestion
  pname = "exoscale-cli";
  version = "1.12.0";
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:12]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:13]: Make sure the sha doesn't change.

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:31]: Aren't you in `maintainers/maintainer-list.nix`?

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:18]: Isn't it the default?

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:31]: this meta key is supposed to be the "downstream" maintainer - you.

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:12]: More complete:

```suggestion
    owner = "exoscale";
    repo = "cli";
    rev = "v${version}";
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:31]: I'm not and that is me. :-)

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:18]: Doesn't look like it - if I remove this it dumps other binaries (doc) in the resulting `bin/`

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:3]: The tool is developed and built with go 1.14 so I wanted to ensure this derivation is in line with that.

**Comment by dramaturg** [comment]
@doronbehar thank you for your patient review - this is my first upstream nix package :-)

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:3]: Since you are the upstream maintainer as well, I think you can be trusted to update this to `buildGo115Package` in the future. But, it seems like an unnecessary maintenance burden to me, as the `buildGoPackage` function uses just the mainline Go compiler available, so if you trust Nixpkgs to always default `go` to the latest `go`, you should be fine with `buildGoPackage`.

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:15]: Yea I encountered such issue as well once... I'd add a comment right above this that explains this. What I did back when I encountered this is:

https://github.com/NixOS/nixpkgs/blob/fa0849d02308ea4d0b646f6fffff07cd042ab810/pkgs/servers/gotify/default.nix#L41-L43

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:19]: What's `-o`? If I didn't understand the comment right away, then perhaps it can be improved :).

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:28]: Well then, 1st of all welcome aboard :). 2ndly, you should add yourself as a maintainer to `maintainers/maintainer-list.nix`, use https://caius.github.io/github_id/ to get your GitHub id, and make sure that eventually whatever the reviews are, you have 2 commits in this PR:

```
maintainers: add dramaturg
exoscale: init at 1.12.0
```

Oh and use:

```nix
    maintainers = with stdenv.lib.maintainers; [ dramaturg ];
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:11]: Mark this conversation as broken, from some reason _I_ can't do that.

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:3]: I want to make sure it's as close as possible to the binaries we publish.
This release was developed and tested with go 1.14 so I'd like to make sure the binary produced by nix does the same. In fact I just realized our readme is not up to date regarding this. :-)

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix]: The bot is not happy with that. Do I need to create a separate PR for that?

File: maintainers/maintainer-list.nix
Comment (doronbehar) [inline maintainers/maintainer-list.nix:2044]:
```suggestion
    githubId = 472846;
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:23]: I see now what this is workarounding. I'm not sure about this. Could you please tell me what exactly did you try and what error / problem you hit? Also, just because I don't know, what does `-o` ?

Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:23]: Ohh never mind, I understand it all now. It may sound like a stupid request, but would it work if you'd use:

```
postInstall = ''
mv $out/bin/cli $out/bin/exo
'';
```

It's just much more intuitive and readable IMO.

Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:23]: `$out` doesn't exist `postInstall` - maybe something with the way go packages are built?

`ls: cannot access '/nix/store/g4ynb45r7w04phmwlw10mgqj0rhz59k8-exoscale-cli-1.12.0': No such file or directory`

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:10]:
```suggestion
    rev    = "v${version}";
```

Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:10]:
```suggestion
    rev    = "v${version}";
```

Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:10]:
```suggestion
    rev    = "v${version}";
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:28]:
```suggestion
    homepage    = "https://github.com/exoscale/cli";
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [inline pkgs/tools/admin/exoscale-cli/default.nix:3]: Needed to use `version = "v${version}"` in `fetchFromGitHub`.


```suggestion
buildGo114Package rec {
```

File: pkgs/tools/admin/exoscale-cli/default.nix
Comment (doronbehar) [reply pkgs/tools/admin/exoscale-cli/default.nix:23]: Yea, the `buildGoPackage` and `buildGoModule` functions are buggy and they are going through some refactoring these days. The comment is pretty good right now so :+1:.

**Comment by doronbehar** [comment]
Sorry for the duplicates ^, but you should be good to go after these.

**Review by doronbehar — APPROVED** [review]

**Pull request merged**
